src/strategies/strategy-univ3-rebalance.sol

```diff
@@ -0,0 +1,560 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.6.12;
+pragma experimental ABIEncoderV2;
+
+import "../lib/erc20.sol";
+import "../lib/safe-math.sol";
+import "../lib/univ3/PoolActions.sol";
+import "../interfaces/jarv2.sol";
+import "../interfaces/uniswapv2.sol";
+import "../interfaces/univ3/IUniswapV3PositionsNFT.sol";
+import "../interfaces/univ3/IUniswapV3Pool.sol";
+import "../interfaces//univ3/IUniswapV3Staker.sol";
+import "../interfaces/univ3/ISwapRouter.sol";
+import "../interfaces/controllerv2.sol";
+import "hardhat/console.sol";
+
+abstract contract StrategyRebalanceUniV3 {
+    using SafeERC20 for IERC20;
+    using Address for address;
+    using SafeMath for uint256;
+    using PoolVariables for IUniswapV3Pool;
+
+    // Perfomance fees - start with 20%
+    uint256 public performanceTreasuryFee = 2000;
+    uint256 public constant performanceTreasuryMax = 10000;
+
+    // User accounts
+    address public governance;
+    address public controller;
+    address public strategist;
+    address public timelock;
+
+    address public univ3_staker;
+
+    // Dex
+    address public univ2Router2 = 0x7a250d5630B4cF539739dF2C5dAcb4c659F2488D;
+    address public sushiRouter = 0xd9e1cE17f2641f24aE83637ab66a2cca9C378B9F;
+    address public constant univ3Factory = 0x1F98431c8aD98523631AE4a59f267346ea31F984;
+    address public constant univ3Router = 0xE592427A0AEce92De3Edee1F18E0157C05861564;
+
+    // Tokens
+    IUniswapV3Pool public pool;
+
+    IERC20 public token0;
+    IERC20 public token1;
+    uint256 public tokenId;
+
+    int24 public tick_lower;
+    int24 public tick_upper;
+    int24 public tickSpacing;
+    int24 public tickRangeMultiplier;
+
+    address public constant weth = 0xC02aaA39b223FE8D0A0e5C4F27eAD9083C756Cc2;
+    address public rewardToken;
+    IUniswapV3PositionsNFT public nftManager = IUniswapV3PositionsNFT(0xC36442b4a4522E871399CD717aBDD847Ab11FE88);
+
+    mapping(address => bool) public harvesters;
+
+    bool public stakingActive = true;
+    IUniswapV3Staker.IncentiveKey key;
+
+    event InitialDeposited(uint256 tokenId);
+    event harvested(uint256 tokenId);
```

Comment:
As a matter of convention, event names are capitalized.

Reply:
Done.